cmd/routedns/main.go

```diff
@@ -229,9 +229,9 @@ func start(opt options, args []string) error {
 			listeners = append(listeners, ln)
 		case "doh":
 			if l.Transport != "quic" {
-				l.Address = rdns.AddressWithDefaultForHttp(l.Address, rdns.DoHPort)
+				l.Address = rdns.AddressWithDefault(l.Address, rdns.DoHPort)
 			} else if l.Transport == "quic" {
-				l.Address = rdns.AddressWithDefaultForHttp(l.Address, rdns.DohQuicPort)
+				l.Address = rdns.AddressWithDefault(l.Address, rdns.DohQuicPort)
 			}
 			tlsConfig, err := rdns.TLSServerConfig(l.CA, l.ServerCrt, l.ServerKey, l.MutualTLS)
```

Comment:
Is this supposed be a change from `AddressWithDefaultForHttp` to `AddressWithDefault` or is that out of date?

Reply:
It's intentional. I changed the function to support both formats, http and regular endpoints. That way you can use the same for both cases. Also added a few tests if you want to confirm this does handle both.

Reply:
Ok